### PR TITLE
Clean up DynamicWinsockMethods, enums, and small structs.

### PR DESCRIPTION
**Pull request by pgavlin**

### CODE REVIEW

**Comment by pgavlin** [comment]
@stephentoub @davidsh PTAL

**Comment by davidsh** [comment]
Left minor comments, otherwise LGTM.

File: src/System.Net.Sockets/src/System/Net/Sockets/SocketFlags.cs [hunk 1]
```diff
@@ -5,78 +5,32 @@
...
-        /// <devdoc>
-        ///    <para>
-        ///       Send without using routing tables.
-        ///    </para>
-        /// </devdoc>
+        // Send without using routing tables.
         DontRoute = 0x0004,
 
-        // see: http://as400bks.rochester.ibm.com/pubs/html/as400/v4r5/ic2978/info/apis/recvms.htm
+        // See: http://as400bks.rochester.ibm.com/pubs/html/as400/v4r5/ic2978/info/apis/recvms.htm
```
Comment (CIPop) [inline src/System.Net.Sockets/src/System/Net/Sockets/SocketFlags.cs:25]: Maybe we should just remove the comment.
[This article](http://blog.stephencleary.com/2010/06/on-lighter-note-socketflagsmaxiovectorl.html) is pointing an interesting fact about it.

**Comment by CIPop** [comment]
LGTM

**Pull request merged**